lib/rucio/core/rule.py

```diff
@@ -890,13 +1029,21 @@ def list_associated_rules_for_file(scope, name, *, session: "Session"):
     """
 
     rucio.core.did.get_did(scope=scope, name=name, session=session)  # Check if the did acually exists
-    query = session.query(models.ReplicationRule).\
-        with_hint(models.ReplicaLock, "INDEX(LOCKS LOCKS_PK)", 'oracle').\
-        join(models.ReplicaLock, models.ReplicationRule.id == models.ReplicaLock.rule_id).\
-        filter(models.ReplicaLock.scope == scope, models.ReplicaLock.name == name).distinct()
+    stmt = select(
+        models.ReplicationRule
+    ).with_hint(
+        models.ReplicationRule, "INDEX(RULES RULES_PK)", 'oracle'
+    ).join(
+        models.ReplicaLock,
+        models.ReplicationRule.id == models.ReplicaLock.rule_id
+    ).where(
+        and_(models.ReplicaLock.scope == scope,
+             models.ReplicaLock.name == name)
+    ).distinct()
+
     try:
-        for rule in query.yield_per(5):
-            yield rule.to_dict()
+        for result in session.execute(stmt).yield_per(5):
+            yield result[0].to_dict()
```

Comment:
This was the potential issue brought up in https://github.com/rucio/rucio/pull/6493#discussion_r1496013829, I decided to keep it like this (without `.scalars()`) in case it is a performance drain, but i am very open to rewrite this to:

`for result in session.execute(stmt).yield_per(5).scalars():`